daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -795,35 +795,46 @@ runLedgerNavigator flags remainingArguments = do
     partyDetails <- Ledger.listParties hostAndPort
 
     withTempDir $ \confDir -> do
-        -- Navigator determines the file format based on the extension so we need a .json file.
-        let navigatorConfPath = confDir </> "navigator-config.json"
+        let navigatorConfPath = confDir </> "ui-backend.conf"
             navigatorArgs = concat
                 [ ["server"]
-                , ["-c", navigatorConfPath]
                 , [host hostAndPort, show (port hostAndPort)]
                 , navigatorPortNavigatorArgs navigatorPort
                 , remainingArguments
                 ]
+
         writeFileUTF8 navigatorConfPath (T.unpack $ navigatorConfig partyDetails)
-        withJar navigatorPath navigatorArgs $ \ph -> do
-            putStrLn "Waiting for navigator to start: "
-            -- TODO We need to figure out a sane timeout for this step.
-            waitForHttpServer (putStr "." *> threadDelay 500000) (navigatorURL navigatorPort)
-            putStr . unlines $
-                [ "Navigator is running at " <> navigatorURL navigatorPort
-                , "Use Ctrl+C to stop."
-                ]
-            exitWith =<< waitExitCode ph
+        unsetEnv "DAML_PROJECT" -- necessary to prevent config contamination
+        withCurrentDirectory confDir $ do
+            withJar navigatorPath navigatorArgs $ \ph -> do
+                putStrLn "Waiting for navigator to start: "
+                -- TODO We need to figure out a sane timeout for this step.
+                waitForHttpServer (putStr "." *> threadDelay 500000) (navigatorURL navigatorPort)
+                putStr . unlines $
+                    [ ""
+                    , "Navigator is running at " <> navigatorURL navigatorPort
+                    , "Use Ctrl+C to stop."
+                    ]
+                exitWith =<< waitExitCode ph
 
   where
     navigatorConfig :: [PartyDetails] -> T.Text
-    navigatorConfig partyDetails =
-        TL.toStrict . encodeToLazyText $ object
-            ["users" .= object
-                [ TL.toStrict displayName .= object [ "party" .= TL.toStrict (unParty party) ]
-                | PartyDetails{..} <- partyDetails
-                ]
+    navigatorConfig partyDetails = T.unlines . concat $
```

Comment:
Why did you change the format here?

Reply:
I *think* the json format doesn't work, but I could be wrong. This might have just been the case of fixing the wrong thing by accident, before fixing the right thing.

But this format definitely works.

Reply:
It definitely used to work at some point and iirc that was a feature of the underlying configuration lib so I would be surprised if it has changed. Using a standard format such as JSON does seem nicer than using whatever weird format that scala config lib is using but given that this will go away with #2418, it’s not worth spending any effort trying to figure out what’s going wrong :slightly_smiling_face:

Reply:
I agree with everything you say :)